### PR TITLE
 ISPN-9587 Key Transformer registration via config

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-9587

### CODE REVIEW

**Comment by anistor** [comment]
This depends on https://github.com/infinispan/infinispan/pull/6312, so I added those commits here too.

**Comment by anistor** [comment]
There was a whitespace related checkstyle violation. Fixed.

**Comment by anistor** [comment]
Green CI. The stars are friendly today :)

**Comment by danberindei** [comment]
Since this didn't make it into 9.4.0, do you want to get it into 9.4.1 or is 10.0 fine? If it's the latter, please change the schema.

**Comment by anistor** [comment]
I set the milestone to 10.0.0.Alpha1 because it adds a new (optional) config element so I'm not sure if such changes are acceptable in a micro.

**Comment by anistor** [comment]
I've updated the asciidocs for this.

**Comment by anistor** [comment]
Rebased again. This should be ready @gustavonalle, want to have a look?

File: documentation/src/main/asciidoc/user_guide/query.adoc [hunk 1]
```diff
@@ -737,15 +738,51 @@ public class CustomTransformer implements Transformer {
...
+   builder.indexing().autoConfig(true)
+         .addKeyTransformer(CustomKey.class, CustomTransformer.class);
+
+----
+
+[small]*Registering a Transformer programmatically at runtime*
+
+Using this technique, you don't have to annotate your custom key type and you also do not add the transformer to the,
+cache indexing configuration, instead, you can add it to the _SearchManagerImplementor_ dynamically at runtime by invoking
+_org.infinispan.query.spi.SearchManagerImplementor.registerKeyTransformer(Class<?>, Class<? extends Transformer>)_:
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/query.adoc]: Ugh, do you really want to document this @anistor ?

**Pull request merged**